server/src/main/java/org/opensearch/extensions/rest/RestInitializeExtensionAction.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.rest;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.extensions.ExtensionDependency;
+import org.opensearch.extensions.ExtensionsManager;
+import org.opensearch.extensions.ExtensionsSettings.Extension;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.rest.RestRequest.Method.POST;
+
+/**
+ * An action that initializes an extension
+ */
+public class RestInitializeExtensionAction extends BaseRestHandler {
+
+    private final ExtensionsManager extensionsManager;
+
+    @Override
+    public String getName() {
+        return ExtensionsManager.REQUEST_EXTENSION_ACTION_NAME;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(POST, "/_extensions/initialize"));
```

Comment:
Following the suggestion what does that `/_extensions/_admin` return?

Reply:
> Following the suggestion what does that `/_extensions/_admin` return?

@peternied right now; an error message that the API isn't available.  Probably a `GET` list of currently installed extensions would make sense.

Reply:
Can we have `/_extensions/shutdown`, `/_extensions/reboot` and rest of the information passed in request body?

Reply:
Re-thinking this, I realized later that we always append an underscore to an extension name, e.g., `/_extensions/_hw`:
https://github.com/opensearch-project/OpenSearch/blob/95135ec32123e0b60d3be90e31b095d62faa3033/server/src/main/java/org/opensearch/rest/extensions/RestSendToExtensionAction.java#L90

So I may retract my suggestion here; as long as we *do not* prepend our commands with an underscore, they work as-is. So I'm okay with this as-is, but this is a good conversation to have.

Reply:
I think we will want to revisit this. It may not matter for Owais' change, but this seems like a recipe for trouble. If we are always prepending extensions with '_' and _admin is going to be linked to administrative commands, then there may be issues if someone were to give their extension the name "admin." You could then grant privileges to call the extension, but run into problems because now you have /extensions/_admin resolving to two things. This would be particularly bad if someone were malicious and could resolve the requests to the admin routes instead of the extension named admin.

Reply:
@scrawfor99 I had already updated my recommendation to note that if we avoid using anything with an underscore starting it, we avoid stepping on any extensions. That gives us the entire dictionary for an API `/_extensions/anything-not-starting-with-underscore/`

Reply:
Sorry about that, I must have misread your comment.